installer.php

```diff
@@ -0,0 +1,105 @@
+$rcfiles = array(                                  // Array of common .rc files to look for.
+  ".bashrc",
+  ".zshrc",
+  ".config/fish/config.fish",
+  ".profile",
+);
+$package = "updatinate";                           // This _should_ be "terminus", but can be changed to test
+                                                   // the script with other packages already configured.
+
+// Function to download Terminus executable file from GitHub to ~/.terminus/bin
+function downloadTerminus($installdir, $package)
+{
+    // $opts defines values required by the GitHub API to respond correclty. $context formats them for use.
+    $opts = [
+            'http' => [
+              'method' => 'GET',
+              'header' => [
+                'User-Agent: PHP'
+              ]
+            ]
+    ];
+    $context  = stream_context_create($opts);
+    $releases = file_get_contents("https://api.github.com/repos/pantheon-systems/" . $package . "/releases", false, $context);
+    $releases = json_decode($releases);
+    $version  = $releases[0]->tag_name;
+    $url      = $releases[0]->assets[0]->browser_download_url;
+    // Do the needful
+    echo("\nDownloading Terminus " . $version . " from " . $url . "\n");
+    $couldDownload = file_put_contents($installdir . "/" . $package . ".phar", file_get_contents($url));
+
+    // Return true if successful
+    return $couldDownload;
+}
+
+
+// Function to add to any common shell configuration files a line to amend $PATH with  ~/.terminus/bin.
+function ammendPath($rcfile, $installdir, &$pathUpdated)
+{
+    $pathUpdated = file_put_contents(getenv('HOME') . "/$rcfile", "# Adds Terminus to \$PATH\nPATH=\$PATH:" . $installdir . "\n\n", FILE_APPEND | LOCK_EX);
+    if (!$pathUpdated) {
+        throw new Exception($rcfile . " found, but unable to write to it.");
+    }
+
+    return $pathUpdated;
+}
+
+// Function to determine if ~/.terminus/bin is already in $PATH
+function checkpath($paths, $installdir)
+{
+
+    if (in_array($installdir, $paths)) {
+        return true;
+    } else {
+        return false;
+    }
+}
+
+// BEGIN ACUTAL DOING OF THINGS!
+
+//Makes ~/.terminus/bin if it doesn't exist.
+if (!file_exists($installdir)) {
+    echo("Creating " . $installdir . "/\n");
+    mkdir($installdir, 0700, true);
+}
+
+//Download terminus.phar
+if (downloadTerminus($installdir, $package)) {
+    echo("Downloaded to " . $installdir . "\n\n");
+} else {
+    echo("Download unsuccessful.\n\n");
+    exit();
+}
+
+// Make Terminus executable
+echo("Making Terminus executable... ");
+chmod($installdir . "/" . $package . ".phar", 0755)
+or exit("\nUnable to set Terminus as executable.\n");
+echo("Done.\n\n");
+
+// If ~/.terminus/bin isn't in path, add it.
+if (checkpath($paths, $installdir) === false) {
+    foreach ($rcfiles as $rcfile) {
+        if (file_exists(getenv('HOME') . "/$rcfile") && is_writable(getenv('HOME') . "/$rcfile")) {
```

Comment:
If multiple rc files exist, this code will update the PATH in all of them. It would probably be better to just pick one.

Reply:
If multiple rc files exist, it's because there are multiple shells on the system. None of the rc files listed are both used by default by any one shell. By updating any it finds, we ensure that Terminus remains in the path even if the user, for example, ditches fish and goes back to ZSH or bash.